### ISSUE TITLE
CMake Errors 

### ISSUE THREAD

**Issue by ZahraAnam** [CMU-Perceptual-Computing-Lab/openpose#1873]
I am very new to using Deep Learning models and trying to run OpenPose on Ubuntu 20.04 with CUDA 10.1 and CuDNN 7.6.5. When I attempt to build my file after Configuring and Generation the following error appears:

CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
CUDA_cublas_LIBRARY (ADVANCED)
    linked by target "caffe" in directory /home/anam/Codes/OpenPose/openpose/3rdparty/caffe/src/caffe

-- Configuring incomplete, errors occurred!
See also "/home/anam/Codes/OpenPose/openpose/build/caffe/src/openpose_lib-build/CMakeFiles/CMakeOutput.log".
See also "/home/anam/Codes/OpenPose/openpose/build/caffe/src/openpose_lib-build/CMakeFiles/CMakeError.log".
make[2]: *** [CMakeFiles/openpose_lib.dir/build.make:107: caffe/src/openpose_lib-stamp/openpose_lib-configure] Error 1
make[1]: *** [CMakeFiles/Makefile2:76: CMakeFiles/openpose_lib.dir/all] Error 2
make: *** [Makefile:84: all] Error 2

Overall the log is as follows:

[ 12%] Performing configure step for 'openpose_lib'
CMake Warning (dev) at cmake/Misc.cmake:32 (set):
  implicitly converting 'BOOLEAN' to 'STRING' type.
Call Stack (most recent call first):
  CMakeLists.txt:25 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found gflags  (include: /usr/include, library: /usr/lib/x86_64-linux-gnu/libgflags.so)
-- Found glog    (include: /usr/include, library: /usr/lib/x86_64-linux-gnu/libglog.so)
-- Found PROTOBUF Compiler: /usr/bin/protoc
-- HDF5: Using hdf5 compiler wrapper to determine C configuration
-- HDF5: Using hdf5 compiler wrapper to determine CXX configuration
-- CUDA detected: 10.1
-- Found cuDNN: ver. 7.6.5 found (include: /usr/include, library: /usr/lib/x86_64-linux-gnu/libcudnn.so)
-- Automatic GPU detection failed. Building for all known architectures.
-- Added CUDA NVCC flags for: sm_35 sm_37 sm_50 sm_52 sm_53 sm_60 sm_61 sm_62 sm_70 sm_72 sm_75 sm_80 sm_86
-- Found Atlas (include: /usr/include/x86_64-linux-gnu library: /usr/lib/x86_64-linux-gnu/libatlas.so lapack: /usr/lib/x86_64-linux-gnu/liblapack.so
-- Python interface is disabled or not all required dependencies found. Building without it...
-- 
-- ******************* Caffe Configuration Summary *******************
-- General:
--   Version           :   1.0.0
--   Git               :   1.0-149-g1807aada
--   System            :   Linux
--   C++ compiler      :   /usr/bin/c++
--   Release CXX flags :   -O3 -DNDEBUG -fPIC -Wall -std=c++11 -Wno-sign-compare -Wno-uninitialized
--   Debug CXX flags   :   -g -fPIC -Wall -std=c++11 -Wno-sign-compare -Wno-uninitialized
--   Build type        :   Release
-- 
--   BUILD_SHARED_LIBS :   ON
--   BUILD_python      :   OFF
--   BUILD_matlab      :   OFF
--   BUILD_docs        :   OFF
--   CPU_ONLY          :   OFF
--   USE_OPENCV        :   OFF
--   USE_LEVELDB       :   OFF
--   USE_LMDB          :   OFF
--   USE_NCCL          :   OFF
--   ALLOW_LMDB_NOLOCK :   OFF
--   USE_HDF5          :   ON
-- 
-- Dependencies:
--   BLAS              :   Yes (Atlas)
--   Boost             :   Yes (ver. 1.71)
--   glog              :   Yes
--   gflags            :   Yes
--   protobuf          :   Yes (ver. 3.6.1)
--   CUDA              :   Yes (ver. 10.1)
-- 
-- NVIDIA CUDA:
--   Target GPU(s)     :   Auto
--   GPU arch(s)       :   sm_35 sm_37 sm_50 sm_52 sm_53 sm_60 sm_61 sm_62 sm_70 sm_72 sm_75 sm_80 sm_86
--   cuDNN             :   Yes (ver. 7.6.5)
-- 
-- Install:
--   Install path      :   /home/anam/Codes/OpenPose/openpose/build/caffe
-- 
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
CUDA_cublas_LIBRARY (ADVANCED)
    linked by target "caffe" in directory /home/anam/Codes/OpenPose/openpose/3rdparty/caffe/src/caffe

-- Configuring incomplete, errors occurred!
See also "/home/anam/Codes/OpenPose/openpose/build/caffe/src/openpose_lib-build/CMakeFiles/CMakeOutput.log".
See also "/home/anam/Codes/OpenPose/openpose/build/caffe/src/openpose_lib-build/CMakeFiles/CMakeError.log".
make[2]: *** [CMakeFiles/openpose_lib.dir/build.make:107: caffe/src/openpose_lib-stamp/openpose_lib-configure] Error 1
make[1]: *** [CMakeFiles/Makefile2:76: CMakeFiles/openpose_lib.dir/all] Error 2
make: *** [Makefile:84: all] Error 2

My problem probably is naive and simple but will be really grateful if someone can help me through it.
Or if there is any Gitter community to help beginners that you can suggest me.

**Comment by mallikagarg** [comment]
Have you resolved the issue, I have got the same.

**Comment by ZahraAnam** [comment]
Yes! By reinstalling the cmake and correct version of CUDA-Toolkit.

**Comment by mallikagarg** [comment]
I have reinstalled cuda-11.1 and cudnn-8.0.4 on ubuntu 20.04 , but still facing the same issue.

**Comment by ZahraAnam** [comment]
Did you deactivate conda??

**Comment by mallikagarg** [comment]
i have not installed anaconda on my system

**Comment by ZahraAnam** [comment]
Hmm! We can connect and check what and where is the problem. For me it worked after setting Cuda

**Comment by xinzi2018** [comment]
i have the same question

**Comment by ZahraAnam** [comment]
Are versions of cuda, opencv etc same as suggested Sent from my Huawei phone-------- Original message --------From: xinzi ***@***.***>Date: Tue., Apr. 20, 2021, 1:09 p.m.To: CMU-Perceptual-Computing-Lab/openpose ***@***.***>Cc: ZahraAnam ***@***.***>, Author ***@***.***>Subject: Re: [CMU-Perceptual-Computing-Lab/openpose] CMake Errors  (#1873)
i have the same question

**Comment by ccl-private** [comment]
> Hmm! We can connect and check what and where is the problem. For me it worked after setting Cuda

@ZahraAnam how do you set your Cuda? 
This is my setting, still failed.
my Cuda version is 10.2, Cudnn is 8.1
![image](https://user-images.githubusercontent.com/36214675/115813018-ffa95f00-a424-11eb-8d7e-53a54fae70f3.png)

**Comment by ccl-private** [comment]
I noticed something, after cmake .. 
the output log shows cuda10.2 is detected by openpose. but for caffe, it uses cuda11? 
![image](https://user-images.githubusercontent.com/36214675/115819117-c8d94600-a430-11eb-80df-fce164daecc6.png)

**Comment by ZahraAnam** [comment]
I have it in a different way. I assign path in bashrc

**Comment by 535091206** [comment]
help
